examples/sparse/sparse_benchmark.ipynb

```diff
@@ -233,16 +261,50 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 8,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Step 1\n",
+      "\t [100, 300]\n",
+      "\t {'time_dense': 1.7728157043457031, 'size_dense': np.int64(100000), 'time_sparse': 3.7954177856445312, 'size_sparse': np.int64(1600)}\n",
+      "Step 2\n",
+      "\t [200, 600]\n",
+      "\t {'time_dense': 1.8389756679534912, 'size_dense': np.int64(400000), 'time_sparse': 3.8383302688598633, 'size_sparse': np.int64(3200)}\n",
+      "Step 3\n",
+      "\t [300, 900]\n",
+      "\t {'time_dense': 1.6056129932403564, 'size_dense': np.int64(900000), 'time_sparse': 3.672597646713257, 'size_sparse': np.int64(4800)}\n",
+      "Step 4\n",
+      "\t [400, 1200]\n",
+      "\t {'time_dense': 2.048386573791504, 'size_dense': np.int64(1600000), 'time_sparse': 3.540794610977173, 'size_sparse': np.int64(6400)}\n",
+      "Step 5\n",
+      "\t [500, 1500]\n",
+      "\t {'time_dense': 1.960519790649414, 'size_dense': np.int64(2500000), 'time_sparse': 3.7853028774261475, 'size_sparse': np.int64(8000)}\n",
+      "Step 6\n",
+      "\t [600, 1800]\n",
+      "\t {'time_dense': 1.60561203956604, 'size_dense': np.int64(3600000), 'time_sparse': 3.637331247329712, 'size_sparse': np.int64(9600)}\n",
+      "Step 7\n",
+      "\t [700, 2100]\n",
+      "\t {'time_dense': 2.128816604614258, 'size_dense': np.int64(4900000), 'time_sparse': 3.795910120010376, 'size_sparse': np.int64(11200)}\n",
+      "Step 8\n",
+      "\t [800, 2400]\n",
+      "\t {'time_dense': 2.252718925476074, 'size_dense': np.int64(6400000), 'time_sparse': 4.16133189201355, 'size_sparse': np.int64(12800)}\n",
+      "Step 9\n",
+      "\t [900, 2700]\n",
+      "\t {'time_dense': 1.7625062465667725, 'size_dense': np.int64(8100000), 'time_sparse': 3.844919443130493, 'size_sparse': np.int64(14400)}\n"
+     ]
+    }
+   ],
    "source": [
     "n_steps = 10\n",
     "\n",
     "res = []\n",
     "for i in range(1, n_steps):\n",
     "    print(f\"Step {i}\")\n",
-    "    num_states = [1000 * i, 3000 * i]\n",
+    "    num_states = [100 * i, 300 * i]\n",
```

Comment:
I had to reduce these to make the benchmark even run on my linux machine without it blowing up the memory